### ISSUE TITLE
Contexts for events

### ISSUE THREAD

**Issue by pmackay** [valueflows/valueflows#238]
Just thinking about how to filter a collection `EconomicEvent` records and wondering if a *context* could be defined/considered that could represent:

* a platform, such as Freecycle
* a place, such as a library
* etc

This seems similar to the context property of an `AgentRelationship` I think? 

An alternative name might be *channel*? *Platform* seems too specific - it works for Freecycle but not a library. 

Any thoughts on this @fosterlynn @bhaugen @elf-pavlik ?

(PS. Great progress on VF recently :-) )

**Comment by fosterlynn** [comment]
Hi @pmackay nice to "see" you!

We have added "scope" to EconomicEvent and Commitment.  That is basically equivalent to "context" on AgentRelationship, although its range is open to more than just Agents.  It is probably not propagated all the way to gitbook yet, so you could look here: https://github.com/valueflows/valueflows/blob/master/release-doc-in-process/all_vf.TTL.

Does that seem like it might work for you?

Please ask more questions if needed!  And of course we're always interested in hearing what you are trying to do. :smile:

**Comment by pmackay** [comment]
That's very interesting, particularly how its described as "The larger boundary or context of an economic event". I have been defining a *context* model that could be a geographic region or an organisation, as "system boundaries" that define the limits of what is being focussed on.

Am wondering if there is any need for a "scope" to define something like Freecycle or library and a "scope" to define a geographic limit, i.e. could more than one be needed? But certainly I think this makes sense :)

Is there any particular reason you named it "scope" for EconomicEvent and Commitment and "context" for Agents?

Thanks @fosterlynn!

**Comment by gcassel** [comment]
I think I like @pmackay 's connection of "context" to "system boundaries", and for "scope" to define something such as geographic limits.    I'm trying to do something similar in Modular Organization Terminology, although the entry for *scope* needs lots of work.   I want (in MOT) to associate *scope* with variables which can limit a clearly possible context-- i.e., systemically related elements-- to a specific active context or "focus".

**Comment by fosterlynn** [comment]
>Is there any particular reason you named it "scope" for EconomicEvent and Commitment and "context" for Agents?

Sort of the twists of history.  We had the "context" for Agents very early on.  Bob and I use that same "context" everywhere in NRP (as context_agent).  @elf-pavlik didn't like the term "context", so we went with "scope" and didn't go back and re-look at the "context" for agent relationships.  And in those discussions, we also said that there might be different kinds of things that could scope a bunch of events, particularly something we don't have defined yet like a project or campaign or maybe a parent process, that isn't really an agent, but contains a lot of processes and could be many deliverables.

It is meant to be kind of loose, to give some flexibility to the way(s) people want to scope their stuff. 

>limits of what is being focussed on

I like that description.

>Am wondering if there is any need for a "scope" to define something like Freecycle or library and a "scope" to define a geographic limit, i.e. could more than one be needed? But certainly I think this makes sense :)

I just looked back to the PR to verify we had the notion of multiple scopes, and we did.  But I don't see some of those changes in the actual documents.  Hmmmm, I need to go back and see what happened!

[edit] Looks OK actually.

**Comment by elf-pavlik** [comment]
currently in: https://github.com/valueflows/valueflows/blob/master/release-doc-in-process/all_vf.TTL#L324-L328 we find
```ttl
vf:context  a        owl:ObjectProperty ;
        rdfs:domain  vf:AgentRelationship ;
        rdfs:label   "context" ;
        rdfs:range   foaf:Agent ;
rdfs:comment "The larger context of a relationship between 2 agents, used where the relationship is not relevant outside of that context." .
```

i don't recall any proposal to open the range of `vf:context` allowing anything as object (`owl:Thing`)

**Comment by fosterlynn** [comment]
>i don't recall any proposal to open the range of vf:context allowing anything as object (owl:Thing)

Right, so that is still `foaf:Agent`.  While vf:scope is `owl:Thing`.  Sorry if I was unclear above??  

@elf-pavlik do you see any problems with how it is defined right now?  If so, let's address it.  Particularly, do you think we should go back and re-look at vf:context for AgentRelationships, or do you think vf:context is different than vf:scope and should stay like it is?

..... and hi, good to see you pop in! :wave:

**Comment by fosterlynn** [comment]
@pmackay and by the way, this is a good page to look at: https://github.com/valueflows/valueflows/blob/master/docs/introduction/scope.md

**Comment by elf-pavlik** [comment]
> ..... and hi, good to see you pop in! :wave: 

I hope to start slowly getting back in the game :smiley_cat: 

I remember that you wanted that relationship between agent can stay defined as only applying in context of another organization (i guess not person). If you don't find it important for `vf:context` property to imply that object in statement using it has `"@type": "foaf:Agent"` we can just use `vf:scope` with open domain and open range - so it has no implications about `"@type"` of subject and object in statements using it.

BTW if we want to have `vf:context` more specific than `vf:scope` we may want to narrow the range to `org:Organization` .

**Comment by fosterlynn** [comment]
>I hope to start slowly getting back in the game :smiley_cat:

woo hoo :+1: :smile:
